api/handlers.go

```diff
@@ -321,6 +362,7 @@ func (api *API) userGET(w http.ResponseWriter, req *http.Request, _ httprouter.P
 
 // userLimitsGET returns the speed limits which apply to this user.
 func (api *API) userLimitsGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	defer api.closeBody(req)
```

Comment:
Do we have to call `closeBody` on all of these? it's easy to forget I think, would be cool if it could be applied automatically. But also, do we have a request body on all of these requests?

Reply:
Yes, we need to always close the body of a request that has a body. Here's a discussion about it:  https://stackoverflow.com/questions/33238518/what-could-happen-if-i-dont-close-response-body

If we don't close it's not a terrible thing but it's a resource leak and, more importantly, it will prevent the HTTP connection from being reused for the next API call the user wants to make. That means another handshake, which is slow. Closing this everywhere does not cost us anything.

Reply:
The stacko you linked is something else in my opinion. There the client is closing it, it does not suggest to close it in the HTTP handler, which I think is actually advised against for example here https://stackoverflow.com/questions/42525499/where-to-put-defer-req-body-close

Reply:
You are right. The server doesn't need to close the body. Removed.